### DISCUSSION TITLE
Without internet, rembg server throws 500 server error

### DISCUSSION THREAD

**Discussion by ericjanofski** [danielgatis/rembg#562] [General]
I'm experiencing a server error when I turn off the internet. I'm using "rembg s". 

I ran it with the internet on and it worked. Wouldn't it download the models it needs at that point? Something requires an internet connection though. If I turn off the internet it throws a 500 server error pointing to issues with asynchio. I'll post the error in a sec.

Can someone assist or let me know if this is a no-go without internet?

**Comment by ericjanofski** [comment]
INFO:     127.0.0.1:54567 - "GET /api/remove?url=http://localhost/test.jpg HTTP/1.1" 200 OK
INFO:     127.0.0.1:54571 - "GET /api/remove?url=http://localhost/test.jpg HTTP/1.1" 500 Internal Server Error
ERROR:    Exception in ASGI application
Traceback (most recent call last):
  File "C:\Users\ejano\AppData\Local\Programs\Python\Python311\Lib\site-packages\aiohttp\connector.py", line 1173, in _create_direct_connection
    hosts = await asyncio.shield(host_resolved)
            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\ejano\AppData\Local\Programs\Python\Python311\Lib\site-packages\aiohttp\connector.py", line 884, in _resolve_host
    addrs = await self._resolver.resolve(host, port, family=self._family)
            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\ejano\AppData\Local\Programs\Python\Python311\Lib\site-packages\aiohttp\resolver.py", line 33, in resolve
    infos = await self._loop.getaddrinfo(
            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\ejano\AppData\Local\Programs\Python\Python311\Lib\asyncio\base_events.py", line 867, in getaddrinfo      
    return await self.run_in_executor(
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\ejano\AppData\Local\Programs\Python\Python311\Lib\concurrent\futures\thread.py", line 58, in run
    result = self.fn(*self.args, **self.kwargs)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\ejano\AppData\Local\Programs\Python\Python311\Lib\socket.py", line 962, in getaddrinfo
    for res in _socket.getaddrinfo(host, port, family, type, proto, flags):
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
socket.gaierror: [Errno 11004] getaddrinfo failed

**Reply by ericjanofski** [reply]
I went to http://localhost:5000/api and hit the "Try it out" button on the get request. It worked the first time with internet connected. I didn't work the second time after I turned off my internet. The response was "Internal Server Error". 

Can someone assist?

**Comment by danielgatis** [comment]
if the models are already downloaded no internet is required.
take a look on this screenshot.

![Screenshot 2024-01-10 at 6 22 42 PM](https://github.com/danielgatis/rembg/assets/7380/80ed3646-c8cf-4450-9aab-602b811be08a)

**Reply by ericjanofski** [reply]
Thanks for the reply. I try the same api code via the browser like this and i get an internal server error again with the same error from above. I think hit has to do with the python aiohttp package. I'm digging deeper on my end. Any insight you can provide would be very helpful, thanks.

**Reply by ericjanofski** [reply]
![image](https://github.com/danielgatis/rembg/assets/24723635/02dbc7eb-04de-4857-806a-eb98715d2f45)

**Reply by ericjanofski** [reply]
Can you share what local server you're using? I've tried wamp and rebex. Neither were successful. I'm wondering if it's some kind of ssl issue with aiohttp, but I'm not sure why it would work on your machine.

I have all firewalls turned off.

**Reply by danielgatis** [reply]
go to your picture folder, and run this command: 

```
python -m http.server
```

after that test if you can access http://your-public-ip:8000/picture.png

**Reply by ericjanofski** [reply]
Thanks, that's a nice utility. Same result though:
aiohttp.client_exceptions.ClientConnectorError: Cannot connect to host localhost:8000 ssl:default [getaddrinfo failed]

**Reply by danielgatis** [reply]
try to use your public IP instead of the "localhost"

**Reply by ericjanofski** [reply]
Ah! Thank you so much Daniel, that did the trick. This worked:

http://localhost:5000/api/remove?url=http://127.0.0.1/test.jpg

I very much appreciate the assistance.
